File: tests/numerical/fit/scripts/bivariate_fit_output.m
% ======= estimation ============================

clc
FLAG_copulaparam = true;  
CopulaFamily = {'Clayton','Frank','Gumbel'};
datasets = {'1','2'};
strMsg = sprintf('Dataset,        Copula, Theta, Tau');
disp(strMsg);
strMsg = sprintf('-----------------------------------');
disp(strMsg);
for d = 1:length(datasets)
    strFile = sprintf('datasets/%s.csv',datasets{d});
    T = readtable(strFile);
    T = table2array(T);
    tau = corr(T(:,1),T(:,2),'Type','Kendall');
    for k=1:length(CopulaFamily)
        if FLAG_copulaparam,
            theta = copulaparam(CopulaFamily{k},tau);
        else
            theta = copulafit(CopulaFamily{k},T);
        end
        strMsg = sprintf('%s, %s, %2.2f, %2.2f',strFile,CopulaFamily{k},theta,tau);
        disp(strMsg);
    end
end

File: tests/numerical/fit/scripts/bivariate_fit_output.R
library(copula)

# Create an empty dataframe
estimation <- data.frame(
    dataset=character(), 
    type=character(), 
    theta=double(), 
    tau=double(),
    stringsAsFactors = FALSE
)

for (dataset in list.files("datasets", full.names=T, pattern="csv")) {
    X = data.matrix(read.csv(dataset))
    
    theta = unname(coef(fitCopula(claytonCopula(), data=X, method = "itau")))
    estimation[nrow(estimation)+1,] <- list(
        dataset,
        "Clayton",
        theta,
        tau(claytonCopula(theta))
    )
    
    theta = unname(coef(fitCopula(frankCopula(), data=X, method = "itau")))
    estimation[nrow(estimation)+1,] <- list(
        dataset,
        "Frank",
        theta,
        tau(frankCopula(theta))
    )
    
    theta = unname(coef(fitCopula(gumbelCopula(), data=X, method = "itau")))
    estimation[nrow(estimation)+1,] <- list(
        dataset,
        "Gumbel",
        theta,
        tau(gumbelCopula(theta))
    )
}

write.csv(estimation, 'estimation.csv', row.names=FALSE)